src/components/WeblnButton.tsx

```diff
@@ -0,0 +1,34 @@
+import log from "loglevel";
+import { Show } from "solid-js";
+
+import { useCreateContext } from "../context/Create";
+import { useGlobalContext } from "../context/Global";
+import { enableWebln } from "../utils/webln";
+
+const WeblnButton = () => {
+    const { t, webln } = useGlobalContext();
+    const { receiveAmount, sendAmountValid, setInvoice } = useCreateContext();
+
+    const createWeblnInvoice = async () => {
+        enableWebln(async () => {
+            const amount = Number(receiveAmount());
+            const invoice = await window.webln.makeInvoice({ amount: amount });
+            log.debug("created webln invoice", invoice);
+            setInvoice(invoice.paymentRequest);
+        });
+    };
+    return (
+        <Show when={webln()}>
```

Comment:
Components themselves should not decide if they are rendered. The parent does that